packages/nvidia_nat_adk/src/nat/plugins/adk/llm.py

```
@@ -194,27 +182,53 @@ async def dynamo_adk(config: DynamoModelConfig, _builder: Builder):
+        from nat.profiler.prediction_trie.trie_lookup import PredictionTrieLookup
+
+        prediction_lookup: PredictionTrieLookup | None = None
+        if config.nvext_prediction_trie_path:
+            try:
+                trie_path = Path(config.nvext_prediction_trie_path)
+                trie = load_prediction_trie(trie_path)
+                prediction_lookup = PredictionTrieLookup(trie)
+                logger.info("Loaded prediction trie from %s", config.nvext_prediction_trie_path)
+            except FileNotFoundError:
+                logger.warning("Prediction trie file not found: %s", config.nvext_prediction_trie_path)
+            except Exception as e:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)